src/controller/subtitle-track-controller.js

```diff
@@ -170,6 +271,46 @@ class SubtitleTrackController extends EventHandler {
     hls.trigger(Event.SUBTITLE_TRACK_LOADING, { url: currentTrack.url, id: trackId });
   }
 
+  /**
+   * @private
+   * Called when we want to reselect the track based on current environment params have been updated,
+   * like wether we should fallback to default selection, or match the current level group id upon
+   * a level switch that involved a group-ID switch for subs.
+   */
+  _selectInitialSubtitleTrack () {
+    // loop through available subtitle tracks and autoselect default if needed
+    // TODO: improve selection logic to handle forced, etc
+
+    let selectedTrack = null;
+
+    const tracks = this.tracks.filter(
+      (track) => this._subtitleGroupId === null || track.groupId === this._subtitleGroupId
+    );
+
+    if (!tracks.length) {
+      return;
+    }
+
+    const defaultTracks = tracks.filter((track) => track.default);
+
+    if (this.selectDefaultTrack && defaultTracks.length) {
+      selectedTrack = defaultTracks[0];
+    } else if (tracks.length) {
+      selectedTrack = tracks[0];
+    } else {
+      logger.warn('No selectable subtitle tracks found');
+      return;
+    }
```

Comment:
This is redundant. We will always have tracks here because of the early return above ( `if (!tracks.length) { return;` ).

```suggestion
    }

    selectedTrack = tracks[0];
```

Reply:
true